### ISSUE TITLE
How to make following shadow? Completely lost.

### ISSUE THREAD

**Issue by ramonsgds** [ZieIony/Carbon#338]
I am trying to understand how to use this library even after having looked into the Wiki pages.

I have the following xml:

 
    <?xml version="1.0" encoding="utf-8"?>
    <carbon.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context="MainView.MainActivity">

    <carbon.widget.ConstraintLayout
        style="@style/carbon_CardView"
        android:layout_width="0dp"
        android:layout_height="100dp"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent">
    </carbon.widget.ConstraintLayout>
    </carbon.widget.ConstraintLayout>

My carbon_CardView style is 

    <style name="carbon_CardView" parent="carbon_ConstraintLayout">
        <item name="android:background">@android:color/white </item>
        <item name="carbon_elevation">10dp</item>
        <item name="carbon_cornerRadius">0.01dp</item>
    </style>

My result is the following:

<img width="376" alt="screen shot 2018-03-19 at 9 35 50 pm" src="https://user-images.githubusercontent.com/13023481/37629446-933b1f62-2bbd-11e8-9ea6-cd5cc459a15e.png">

I want to achieve the following:

<img width="387" alt="screen shot 2018-03-19 at 9 40 00 pm" src="https://user-images.githubusercontent.com/13023481/37629541-1ddf6ca4-2bbe-11e8-9923-3a18094549d4.png">

As you can see, I need the shadow to be way more subtle than it is. The way it looks as of now, it's identical to a simple `elevation=20dp` on the standard API 21+ library. How can I achieve the latter with Carbon?

**Comment by ZieIony** [comment]
TL;DR: you can only change the elevation. If that doesn't help, then you can't.

Longer explanation:

On API >= 21 Carbon uses the built-in shadow drawing mechanism. For example, 'app:carbon_elevation' is translated directly to 'android:elevation'. There's no way of changing any of parameters of that mechanism.

On API < 21 Carbon uses the built-in gaussian blur shader. It can be tweaked, but the library doesn't provide any parameters for that. You would have to clone the library and play with the classes from carbon.shadow package. The problem with that blur shader is that it supports only 25 samples, which is not enough to make such soft shadow on very dense screens. Carbon tries to overcome that by using scaled bitmaps - you would have to do that as well or rewrite the shader which would most likely affect performance.

It's also worth noting, that the native/support shadow may be different on different phones and APIs due to vendor-specific tweaks. I'm pretty sure that Samsung changed something in the implementation, because the last time I compared the shadows, on S7 it had a different look.

**Comment by jemshit** [comment]
@ZieIony what about position (dx,dy offset) of shadow, is that configurable?

**Comment by ZieIony** [comment]
No. All parameters are generated automatically based on view's position (x,y,z). Just like in the official implementation. Messing with position could lead to visual artifacts as Carbon's shadows are, simply speaking, 9-patches with empty center.
